### PR TITLE
Add Wiki Pages on Custom Enemies

### PR DESCRIPTION
**Pull request by Hamunii**
Implementing my guide on custom enemies to the wiki. Currently it's added under LethalLib, but I don't think this makes a lot of sense, since LethalLib is only a small part of the project itself.

There is also duplicate information regarding the creation of asset bundles on the unity page, which I've marked with a red `DUPLICATE INFORMATION` container.

The guide is not quite 100% done, but it's getting there. There also might be issues I've not noticed when porting the guide here. Also I guess it should be noted that the guide is more about how to work with the example enemy project / how we got there, rather than a tutorial you follow along to make something from scratch.

Also, how do we want to deal with the repository for the example enemy? We could make a repository under the Lethal Company Community organization and change the enemy from Toilet Leech to something more neutral.

### CODE REVIEW

File: docs/dev/apis/lethallib/custom-enemies/blender.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+---
+prev: true
+next: true
+description: A tutorial on how to add custom enemies by using LethalLib.
```
Comment (Xilophor) [inline docs/dev/apis/lethallib/custom-enemies/blender.md:4]: Change the description to be a little more specific

File: docs/dev/apis/lethallib/custom-enemies/blender.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+**Animations are broken in Unity.**
+This might be because you have animations with the same name in your NLA editor. Make sure your animations have unique names.
+
+## Exporting Assets For Unity
+
+To export your model, go to: `File` -> `Export` -> `FBX (.fbx)`  
+This will open our FBX exporter window, where we have some options available to us. The most important thing here however is the transform section. Because of the differences in Blender's and Unity's coordinate systems, exporting your model is not quite as straight-forward as you'd think. It's very easy to get your model pointing in the wrong direction, being sideways, or even upside down if you don't have correct values set.
+
+Even I don't understand how any it really works, but our model points in the negative Z direction, and in the transfrom section, we have set `Forward` to `-Z Forward`, and `Up` to `Y Up` and our model appears correctly in Unity.
```
Comment (Xilophor) [inline docs/dev/apis/lethallib/custom-enemies/blender.md:78]: Unity uses a left-handed, Y-Up coordinate system, whereas Blender uses a right-handed, Z-Up coordinate system. The transformation you do changes Blender's frame of reference to match the coordinate system Unity uses.

File: docs/dev/apis/lethallib/custom-enemies/coding-ai.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
+---
+prev: true
+next: false
+description: A tutorial on how to add custom enemies by using LethalLib.
```
Comment (Xilophor) [inline docs/dev/apis/lethallib/custom-enemies/coding-ai.md:4]: Make the description a little more specific

File: docs/dev/apis/lethallib/custom-enemies/unity.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
+---
+prev: true
+next: true
+description: A tutorial on how to add custom enemies by using LethalLib.
```
Comment (Xilophor) [inline docs/dev/apis/lethallib/custom-enemies/unity.md:4]: Please make the description a little more specific

File: docs/.vitepress/config.mts [hunk 5]
```diff
@@ -122,6 +122,17 @@ export default defineConfig({
...
                       items: [
                         { text: 'Custom Scrap', link: '/dev/apis/lethallib/customscrap'},
+                        {
+                          text: 'Custom Enemies',
+                          link: '/dev/apis/lethallib/custom-enemies/about',
+                          collapsed: true,
+                          items: [
+                            { text: '3D modeling with Blender', link: '/dev/apis/lethallib/custom-enemies/blender'},
+                            { text: 'Unity project', link: '/dev/apis/lethallib/custom-enemies/unity'},
+                            { text: 'Coding our AI', link: '/dev/apis/lethallib/custom-enemies/coding-ai'},
```
Comment (Xilophor) [inline docs/.vitepress/config.mts:132]: Change the titles here to Title Case

File: docs/dev/apis/lethallib/custom-enemies/coding-ai.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+---
+# Coding our custom AI
```
Comment (Xilophor) [inline docs/dev/apis/lethallib/custom-enemies/coding-ai.md:6]: Please use Title Case

**Review by Xilophor — COMMENTED** [review]
A few suggestions so far

File: docs/dev/apis/lethallib/custom-enemies/unity.md [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+### Adding Items To An Asset Bundle
+
+::: tip
+For information about Asset Bundles, see [Asset Bundling](/dev/intermediate/asset-bundling).
+:::
+
+::: danger DUPLICATE INFORMATION
+This section of the wiki page has duplicate information regarding the creation of Asset Bundles.
+
+**Note:** The existing page on the creation of asset bundles is slightly less in-depth, and does mention how to embed asset bundles in DLL files, which is what we do in this project.
```
[the remark this answers is not in the document]
Comment (Hamunii) [reply docs/dev/apis/lethallib/custom-enemies/unity.md:135]: The "slightly less in-depth" mostly refers to that I mention about items auto adding to the asset bundle, but still having to explicitly include the ones you refer to in the code. Also I guess I do prefer more images as that means a bit less time spent searching around the UI, but yeah technically that part doesn't add much.

Also I had no idea about the loading into memory twice issue, so yeah I'll make the project not do embedded asset bundles. We absolutely need this info in the asset bundle guide

File: docs/dev/apis/lethallib/custom-enemies/unity.md [hunk 8]
```diff
@@ -0,0 +1,160 @@
+---
+prev: true
+next: true
+description: A tutorial on how to add custom enemies by using LethalLib.
+---
+# Unity
+
+You can open the Unity project included in the repository by choosing to open a project from disk, and selecting the `UnityProject` folder. When Unity has loaded the project, look into the ToiletLeech folder for the assets that make up the asset bundle.
```
[the remark this answers is not in the document]
Comment (Hamunii) [reply docs/dev/apis/lethallib/custom-enemies/unity.md:8]: Yeah I also want to do this, but I want a different repository for a dedicated example enemy (even just by renaming Toilet Leech to ExampleEnemy), as toilet leech has just basically been me getting the enemy working in the first place. I guess I could make the repo on my account, but also it would make a lot of sense for it to be by the lethal company community organization as I stated in my original message, as the guide will be on the wiki.

[the remark this answers is not in the document]
Comment (Hamunii) [reply docs/dev/apis/lethallib/custom-enemies/unity.md]: Done! I also made an LC-ExampleEnemy repository.

**Comment by Hamunii** [comment]
All above issues should be fixed now. Also now the guide links to https://github.com/Hamunii/LC-ExampleEnemy instead of ToiletLeechIsReal.

**Comment by Hamunii** [comment]
The guide is now up to date.

File: docs/dev/apis/lethallib/custom-enemies/overview.md [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+You can install Blender from https://www.blender.org/, and it's also available on Flathub for Linux users.
+
+For getting started with making your 3D models in Blender, see [3D Modeling With Blender](./blender-resources.md).
+
+## Unity
+
+Lethal Company uses Unity version 2022.3.9f1, and therefore we should use it too in order to avoid any issues with version differences when exporting our asset bundles.  
+
+::: warning IMPORTANT
+On Linux, Unity 2022.3.9f1 has a bug where it might output audio to the wrong audio device. If this happens, you can try using [PulseAudio Volume Control](https://flathub.org/apps/org.pulseaudio.pavucontrol) to change Unity's audio output (appears as `FMOD Ex App`) to the correct one.
```
[the remark this answers is not in the document]
Comment (Hamunii) [reply docs/dev/apis/lethallib/custom-enemies/overview.md:35]: If you are adding sounds to your enemy and would be debugging the issue of no sounds for a couple of hours otherwise, then this is very relevant.

File: docs/dev/apis/lethallib/custom-enemies/coding-ai.md [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+next: false
+description: An introduction to coding your own enemy AI.
+---
+# Coding Our Custom AI
+
+::: warning IMPORTANT
+You should use a [decompiler](/dev/initial-setup#decompiler) to look how the enemy AI scripts work in the game. All the game code is contained inside `Lethal Company/Lethal Company_Data/Managed/Assembly-CSharp.dll`, so you should open that file in your decompiler!
+:::
+
+## Tips For Testing Your Mod
```
[the remark this answers is not in the document]
Comment (Hamunii) [reply docs/dev/apis/lethallib/custom-enemies/coding-ai.md:12]: I also thought of this, but yeah I don't really know where else it should be. I guess the best would be to have an entirely new page for it. I am also working on a tool that I call `TestingLib` to help with debugging enemies and just speeding up the testing process in general, so there is definitely more to add to such page in the future.

File: docs/dev/apis/lethallib/custom-enemies/blender-resources.md [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+
+- [Fast Character Modeling with the Skin Modifier in Blender](https://youtu.be/DAAwy_l4jw4) - Joey Carlino
+    - Introduces a super cool and easy modeling technique. I recommend this a lot for learning to make basic initial meshes for characters or creatures.
+
+
+### Modeling - Common Issues
+
+**My mesh looks inverted in Blender or when imported to Unity.**  
+This is because your normals got inverted in one way or another. Select your mesh in Edit Mode, press A to select everything, press `Shift+N` to recalculate your normals (do not select "Inside", that is the flipped state).  
+If this doesn't fix the problem after importing to Unity, you likely have resized your object by a negative amount. This looks normal in Blender, but not in Unity. To fix this, go into Object Mode, select your object, press `Ctrl+A`, select apply scale. Now your normals should have flipped in Blender. Now, recalculate normals.
```
Comment (Xilophor) [inline docs/dev/apis/lethallib/custom-enemies/blender-resources.md:38]: Change this section to to reflect a Q/A section, as in
```md
**Q: issue a**

> first paragraph
> <br>
> second paragraph
```

File: docs/dev/apis/lethallib/custom-enemies/blender-resources.md [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+## Modeling
+
+- [Fast Character Modeling with the Skin Modifier in Blender](https://youtu.be/DAAwy_l4jw4) - Joey Carlino
+    - Introduces a super cool and easy modeling technique. I recommend this a lot for learning to make basic initial meshes for characters or creatures.
+
+
+### Modeling - Common Issues
+
+**My mesh looks inverted in Blender or when imported to Unity.**  
+This is because your normals got inverted in one way or another. Select your mesh in Edit Mode, press A to select everything, press `Shift+N` to recalculate your normals (do not select "Inside", that is the flipped state).
```
Comment (Xilophor) [inline docs/dev/apis/lethallib/custom-enemies/blender-resources.md:37]: surround the key A in backticks so it looks like
```md
..., press `A` to select...
```

File: docs/dev/apis/lethallib/custom-enemies/blender-resources.md [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+- [Character animation for impatient people - Blender Tutorial](https://youtu.be/GAIZkIfXXjQ) - Joey Carlino
+    - If you don't want to make and rig your own models.
+
+### Animation - Common Issues
+
+**Objects in my model appear in different places as in Blender when exporting to Unity.**  
+This might be because you have directly animated an object, instead of an armature. Try parenting your object to an armature and remake your animation with that.
+
+**Animations are broken in Unity.**  
+This might be because you have animations with the same name in your NLA editor. Make sure your animations have unique names.
```
Comment (Xilophor) [inline docs/dev/apis/lethallib/custom-enemies/blender-resources.md:82]: Same formatting issue as before

File: docs/dev/apis/lethallib/custom-enemies/blender-resources.md [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+
+![Screenshot: Export as FBX Settings](/images/lethallib/custom-enemies/blender/BlenderExportAsFBX.png)
+
+The most important thing here however is the transform section. Because of the differences in Blender's and Unity's coordinate systems, exporting your model is not quite as straight-forward as you'd think. It's very easy to get your model pointing in the wrong direction, being sideways, or even upside down if you don't have correct values set.
+
+Blender considers "Forward" to be `-Y`, while in Unity it's `Z`. And "Up" in Blender is `Z`, while in Unity it's `Y`. "Forward" is illustrated with the green arrow in the image below, while the blue arrow is "Up". The left side represents Blender's coordinate space, center represents our export settings in Blender's space, while the right side represents the output in Unity. We set `Forward` (`-Y`) to `-Z Forward`, and `Up` (`Z`) to `Y Up` and our model appears correctly in Unity space. The reason `-Z Forward` becomes positive `Z` might be because technically, `Z` is inverted between Blender and Unity, which we can see in the image.
```
Comment (Xilophor) [inline docs/dev/apis/lethallib/custom-enemies/blender-resources.md:93]: Consider splitting this paragraph into two lines to make it easier to read (you just need to add two new lines at somepoint in the paragraph)

File: docs/dev/apis/lethallib/custom-enemies/blender-resources.md [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+
+The most important thing here however is the transform section. Because of the differences in Blender's and Unity's coordinate systems, exporting your model is not quite as straight-forward as you'd think. It's very easy to get your model pointing in the wrong direction, being sideways, or even upside down if you don't have correct values set.
+
+Blender considers "Forward" to be `-Y`, while in Unity it's `Z`. And "Up" in Blender is `Z`, while in Unity it's `Y`. "Forward" is illustrated with the green arrow in the image below, while the blue arrow is "Up". The left side represents Blender's coordinate space, center represents our export settings in Blender's space, while the right side represents the output in Unity. We set `Forward` (`-Y`) to `-Z Forward`, and `Up` (`Z`) to `Y Up` and our model appears correctly in Unity space. The reason `-Z Forward` becomes positive `Z` might be because technically, `Z` is inverted between Blender and Unity, which we can see in the image.  
+![Illustration: Blender coordinates to Unity](/images/lethallib/custom-enemies/blender/BlenderToUnityCoordinates.png)
```
Comment (Xilophor) [inline docs/dev/apis/lethallib/custom-enemies/blender-resources.md:94]: Add an empty line above this to add some spacing between the text and the image

File: docs/dev/apis/lethallib/custom-enemies/overview.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (MaxWasUnavailable) [reply docs/dev/apis/lethallib/custom-enemies/overview.md:35]: Could you slightly reword this to make it apparent as to why this is relevant (e.g. something like your reply above)?

File: docs/dev/apis/lethallib/custom-enemies/unity-project.md [hunk 16]
```diff
@@ -0,0 +1,132 @@
...
+2. Network Object
+    - Needs to be added so our enemy's position can sync in multiplayer. After you reference your AI script, Unity will automatically prompt you to add this component.
+3. Nav Mesh Agent
+    - Allows our enemy to act as a nav mesh agent, which is Unity's system for making easy pathfinding in 3D with the help of a nav mesh that the agents walk on.
+4. Animator
+    - This allows us to control the animations of our model. This deserves its own section, and I barely know anything about Unity's animation system.
+
+We also have these as children of the prefab itself:
+1. ScanNode
+    - Allows us to scan the enemy. Make sure the following is set: Tag: `DoNotSet`, Layer: `ScanNode`
```
Comment (Xilophor) [inline docs/dev/apis/lethallib/custom-enemies/unity-project.md:94]: separate Tag & Layer into sub-points like so
```md
1. a
    - stuff
        - tag
        - layer
```

File: docs/dev/apis/lethallib/custom-enemies/unity-project.md [hunk 17]
```diff
@@ -0,0 +1,132 @@
...
+2. MapDot
+    - Allows us to see the enemy on map. Make sure the following is set: Tag: `DoNotSet`, Layer: `MapRadar`
```
Comment (Xilophor) [inline docs/dev/apis/lethallib/custom-enemies/unity-project.md:96]: same as above

File: docs/dev/apis/lethallib/custom-enemies/unity-project.md [hunk 18]
```diff
@@ -0,0 +1,132 @@
...
+3. Collision
+    - Has the following components: Enemy AI Collision Detect (Script) & Box Collider with `isTrigger: true`
```
Comment (Xilophor) [inline docs/dev/apis/lethallib/custom-enemies/unity-project.md:98]: same as above but with the script & the collider

File: docs/dev/apis/lethallib/custom-enemies/coding-ai.md [hunk 19]
```diff
@@ -0,0 +1,209 @@
...
+
+Before we get into coding our custom AI, it's a good idea to optimize our testing setup so we'll waste less time waiting in loading screens and wandering around.
+
+For opening the game faster, you can open the game directly from the exe file. This means the game is not run through Steam and we must use LAN mode. This also allows us to open two instances of the game and test multiplayer in LAN mode.
+
+For entering a game quickly, we can use [DevelopmentStartup](https://thunderstore.io/c/lethal-company/p/CTNOriginals/DevelopmentStartup/). *Do note that it does not necessarly support joining local games on a second instance yet.*
+
+We can also use [LethalDevMode](https://thunderstore.io/c/lethal-company/p/megumin/LethalDevMode/) to get access to the game's internal debug tools. These tools allow you to toggle a testing level from the in-game menu without having to land your ship, and also spawning your enemy, including toggling invincibility and spawning items.
+![Screenshot: Toggle Testing Room](/images/lethallib/custom-enemies/coding-ai/DebugToggleTestRoom.png)
```
Comment (Xilophor) [inline docs/dev/apis/lethallib/custom-enemies/coding-ai.md:21]: Add empty line above the image for spacing

**Review by Xilophor — CHANGES_REQUESTED** [review]
Overall lgtm, just some minor formatting issues.

File: docs/dev/apis/lethallib/custom-enemies/blender-resources.md (hunk 14 above)
Comment (Hamunii) [reply docs/dev/apis/lethallib/custom-enemies/blender-resources.md:93]: I moved a part of the text below the image instead.

File: docs/dev/apis/lethallib/custom-enemies/coding-ai.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (Xilophor) [reply docs/dev/apis/lethallib/custom-enemies/coding-ai.md:12]: Yeah, I say separate this into its own PR and put it in the `Creating Mods` section.

**Review by MaxWasUnavailable — APPROVED** [review]
